File: LV2/2.8/guess/guess.go
package main

import "fmt"

func main() {
	l := 1
	r := 1001
	var ans string

	for {
		m := (l + r) / 2
		fmt.Println(m)
		fmt.Scanln(&ans)

		if ans == "correct" {
			break
		}

		if ans == "lower" {
			r = m
		} else {
			l = m
		}
	}
}

File: LV2/2.8/guess/guess.rb
l = 1
r = 1001

loop do
  m = (l + r) / 2
  puts m
  STDOUT.flush
  ans = gets.chomp

  break if ans == 'correct'

  if ans == 'lower'
    r = m
  elsif ans == 'higher'
    l = m
  end
end
